### ISSUE TITLE
no-var fixed to incorrect code

### ISSUE THREAD

**Issue by aladdin-add** [eslint/eslint#11441]
**Tell us about your environment**

* **ESLint Version:** v5.14.1/master
* **Node Version:** n/a
* **npm Version:** n/a

**What parser (default, Babel-ESLint, etc.) are you using?**
`@typescript-eslint/parser`
**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{rules: {no-var: 2}}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
declare var __IS_PRODUCTION__: boolean;
```

```bash

```

**What did you expect to happen?**
not fixed to incorrect code

**What actually happened? Please include the actual, raw output from ESLint.**
```ts
let var __IS_PRODUCTION__: boolean;
```

**Are you willing to submit a pull request to fix this bug?**
yes

**Comment by platinumazure** [comment]
This should be pretty easy-- instead of fixing the first token in the VariableDeclaration node, we should fix the first token that specifically says `var`. `sourceCode.getFirstToken` supports a filter predicate.

The tricky bit will be writing a test case with a fixture. :smile:

Marking as accepted.

**Comment by aladdin-add** [comment]
> The tricky bit will be writing a test case with a fixture.

The question is, we are not using typescript for parsing. what do you think of just not autofix this case if  the first token is not `var`?

**Comment by aladdin-add** [comment]
Another issue here is: I remember (not very sure) eslint will not apply the fix, if the fixed code not parsing successfully.
But it seems not working as expected.

**Comment by platinumazure** [comment]
> The question is, we are not using typescript for parsing. what do you think of just not autofix this case, if the first token is not `var`? in this way -- it could be seen as a guard, not a bugfix.

I think if we just use something like `sourceCode.getFirstToken(node, t => t.value === "var")`, we could treat this as a bug fix and we wouldn't need to suppress the autofix. (I left a note with a little more detail in the PR.)
